### ISSUE TITLE
coplanar_waveguide accepts a complex ep_r, drops the loss, and returns a lossless S-matrix

### ISSUE THREAD

**Issue by Alisama20** [flaport/sax#124]
**Version:** reproduced on sax 0.18.2 (current release) and 0.16.14, jax 0.9.2

## What happens

`sax.models.rf.coplanar_waveguide` accepts a complex `ep_r`, discards the
imaginary part, and returns the S-matrix of a **lossless** line. The dielectric
loss vanishes and the result still looks like a valid answer.

```python
import numpy as np
from sax.models.rf import coplanar_waveguide

f = np.array([1e9, 10e9, 100e9])
kw = dict(f=f, length=800.0, width=20.0, gap=15.0, substrate_thickness=100.0)

s = coplanar_waveguide(**kw, ep_r=11.9 * (1 - 0.01j))   # tand = 0.01
print(np.abs(s[("o1", "o2")]))
```

```
[1. 1. 1.]
```

`|S21| = 1` at every frequency, for a substrate that was specified as lossy.
The correct call, `coplanar_waveguide(**kw, ep_r=11.9, tand=0.01)`, gives
`[0.99980478 0.99804949 0.98066523]`.

## Why it is easy to miss

A `ComplexWarning` is emitted, but under default warning filters it fires
**once per call site** and then never again:

```python
for i in range(5):
    cpw_epsilon_eff(20.0, 15.0, 100.0, 11.9 * (1 - 0.01j))
# warning on the first call only; calls 2-5 are silent
```

In a frequency sweep that is one line at the top of a long log. Under
`jax.jit` the value is still truncated and returned as `float64`, with no
error.

## Where

`src/sax/models/rf.py` @ 0.18.2 casts the permittivity to real in nine places:

- line 509 `cpw_epsilon_eff`, `ep_r`
- lines 543, 592 `cpw_z0` / `cpw_z0_thick`, `ep_eff`
- line 644 `microstrip_epsilon_eff`, `ep_r`
- line 697 `microstrip_z0`, `ep_eff`
- lines 757-758 `microstrip_z0_thick`, `ep_r` and `ep_eff`
- lines 834-836 `propagation_constant`, `ep_eff`, `tand`, `ep_r`

all of the form `jnp.asarray(x, dtype=float)`.

## Note on scope

The signatures are annotated `sax.FloatArrayLike`, so real-only input is
clearly the intent, and `tand` is the supported way to express dielectric loss.
`tand` works and also accepts an array over frequency, so nothing is missing
functionally — I checked that before filing:

```python
coplanar_waveguide(**kw, ep_r=11.9, tand=np.array([0.001, 0.010, 0.030]))
# -> |S21| = [0.99998048 0.99804949 0.94310995]
```

So this is not a request for complex-permittivity support. It is about what
happens when someone passes one anyway: the public model accepts the value and
returns a physically wrong answer instead of refusing.

## Suggested fix

Reject complex input where it enters, with a message pointing at `tand`, so it
fails loudly rather than warning once and returning a lossless result. Roughly:

```python
if jnp.iscomplexobj(ep_r):
    msg = "ep_r must be real; use tand to specify dielectric loss"
    raise TypeError(msg)
```

Happy to send a PR for this if you agree with the approach - it is small and
mechanical, and I would add a test that the lossy and lossless calls no longer
return the same S-matrix silently.

## How I ran into it

Prototyping a rational/pole-residue fit of CPW dispersion for
[gsim#228](https://github.com/gdsfactory/gsim/issues/228) item 6. I added a
Debye relaxation to the substrate to check the method against a genuinely
dispersive material; the fit failed and the error grew with more samples,
because the loss never reached the formula. That cost a debugging cycle, which
is why I think failing loudly is worth it.
